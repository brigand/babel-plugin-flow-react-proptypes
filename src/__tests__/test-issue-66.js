const babel = require('babel-core');
const content = `

// @flow

export default function(
  url: string,
  options: PhenomicStaticConfig,
  Html: Function = DefaultHtml
): Promise<string> {

  return <div/>;
}


`;

it('issue 66', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
