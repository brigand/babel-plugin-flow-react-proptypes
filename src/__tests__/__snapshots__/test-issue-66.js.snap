exports[`test issue 66 1`] = `
"\"use strict\";

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

exports.default = function (url, options) {
  var Html = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : DefaultHtml;


  return React.createElement(\"div\", null);
};"
`;
